packages/ember-routing/tests/system/router_test.js

```diff
@@ -255,3 +255,57 @@ QUnit.test('Router#triggerEvent ignores handlers that have not loaded yet', func
 
   triggerEvent(handlerInfos, false, ['loading']);
 });
+
+QUnit.test('Router#_queryParamsFor returns the cached leaf handler when present', function(assert) {
```

Comment:
I would rather these tests be done as acceptance-type tests rather than unit tests. They should probably go in this test file which does the other collision checks: https://github.com/emberjs/ember.js/blob/master/packages/ember/tests/routing/query_params_test/overlapping_query_params_test.js